integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -500,29 +500,25 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
-@pytest.mark.parametrize('ansi_mode', ['nonAnsi', 'ansi'])
-@pytest.mark.parametrize('exp_type', ['bothLiterals', 'justZeroLiteral', 'noLiterals'])
-def test_div_by_zero(ansi_mode, exp_type):
-    if ansi_mode == 'ansi':
-        if is_before_spark_310():
-            pytest.xfail('https://github.com/apache/spark/pull/29882')
-        elif exp_type != 'bothLiterals':
-            pytest.xfail('https://github.com/NVIDIA/spark-rapids/issues/1464')
-
+@pytest.mark.parametrize('ansi_mode, expr', [
+    pytest.param('ansi', '1/0', marks=pytest.mark.xfail(condition=is_before_spark_310(),
```

Comment:
Nit: Should we move the condition to within the test?  It would be less verbose than repeating it three times here.  If we could annotate the test only once with this condition that would work too.

Reply:
I changed this because my first attempt was mixing up reasons when done in the test, and generally did not adhere to https://docs.pytest.org/en/latest/skipping.html#skip-xfail-with-parametrize. But I can split the test cases to simplify.